sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/client/XdsClient.java

```diff
@@ -0,0 +1,121 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.implement.service.xds.client;
+
+import io.envoyproxy.envoy.service.discovery.v3.AggregatedDiscoveryServiceGrpc;
+import io.envoyproxy.envoy.service.discovery.v3.AggregatedDiscoveryServiceGrpc.AggregatedDiscoveryServiceStub;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.MetadataUtils;
+import io.grpc.stub.StreamObserver;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.config.ConfigManager;
+import io.sermant.core.service.xds.config.XdsConfig;
+import io.sermant.core.utils.FileUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.net.ssl.SSLException;
+
+/**
+ * XdsClient
+ *
+ * @author daizhenyu
+ * @since 2024-05-09
+ **/
+public class XdsClient implements Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    private XdsConfig config = ConfigManager.getConfig(XdsConfig.class);
+
+    private ManagedChannel channel;
+
+    /**
+     * construction method
+     */
+    public XdsClient() {
+        createChannel();
+    }
+
+    private void createChannel() {
+        if (config.isSecurityEnable()) {
+            try {
+                SslContext context = GrpcSslContexts.forClient()
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                channel = NettyChannelBuilder.forTarget(config.getControlPlaneAddress())
+                        .sslContext(context)
+                        .build();
+            } catch (SSLException e) {
+                LOGGER.log(Level.SEVERE, "SSLException occurred when creating gRPC channel", e);
+            }
+        } else {
+            channel = NettyChannelBuilder.forTarget(config.getControlPlaneAddress())
+                    .usePlaintext()
+                    .build();
+        }
+    }
+
+    /**
+     * update channel
+     */
+    public void updateChannel() {
+        if (channel != null && !channel.isShutdown() && !channel.isTerminated()) {
+            return;
+        }
+        synchronized (this) {
+            if (channel != null && !channel.isShutdown() && !channel.isTerminated()) {
+                return;
+            }
+            createChannel();
+        }
```

Comment:
use `||` instead of `&&` to make it more clear

Reply:
done